app/lib/features/space/pages/chats_page.dart

```diff
@@ -99,18 +99,7 @@ class SpaceChatsPage extends ConsumerWidget {
                   child: ConvoCard(
                     room: rooms[index],
                     showParent: false,
-
-                    /// FIXME: push is broken for switching from subshell to subshell
-                    /// hence we are using `go` here.
-                    /// https://github.com/flutter/flutter/issues/125752
-                    onTap: () => isDesktop
-                        ? context.goNamed(
-                            Routes.chatroom.name,
-                            pathParameters: {
-                              'roomId': rooms[index].getRoomIdStr(),
-                            },
-                          )
-                        : context.pushNamed(
+                    onTap: () => context.goNamed(
```

Comment:
How certain are we that this would allow us to backtrack without route breaking ?

Reply:
As we are using stateful routing where each tabs are preserving their own routing stage, we can go use goNamed to select specific tab shell selected.

When user clicks on back then it will open that particular shell's back route however over tabs routes are still presevered.

If we use pushNamed then it will push screen of different shell on it's own shell (selected shell).

I have discuss about this with @gnunicorn as well.